contrib/mesos/pkg/scheduler/podtask/predicate.go

```diff
@@ -57,12 +58,10 @@ func NodeSelectorPredicate(t *T, offer *mesos.Offer) bool {
 
 	// check the NodeSelector
 	if len(t.Pod.Spec.NodeSelector) > 0 {
-		slaveLabels := map[string]string{}
-		for _, a := range offer.Attributes {
-			if a.GetType() == mesos.Value_TEXT {
-				slaveLabels[a.GetName()] = a.GetText().GetValue()
-			}
-		}
+		// TODO(sttts): watch nodes and add any other custom labels to this predicate here
```

Comment:
It's not super critical. But if the user sets labels manually (see comment below), our scheduler does not see them. I started to implement that. Shouldn't be too hard.